hushline/premium.py

```diff
@@ -0,0 +1,609 @@
+            return redirect(url_for("premium.index"))
+
+        # Create a Stripe Checkout session
+        try:
+            checkout_session = stripe.checkout.Session.create(
+                client_reference_id=str(user.id),
+                customer=user.stripe_customer_id,
+                line_items=[{"price": business_tier.stripe_price_id, "quantity": 1}],
+                mode="subscription",
+                success_url=url_for("premium.waiting", _external=True),
+                automatic_tax={"enabled": True},
+                customer_update={"address": "auto"},
+            )
+        except stripe._error.StripeError as e:
+            current_app.logger.error(f"Failed to create Stripe Checkout session: {e}")
+            return abort(500)
+
+        if checkout_session.url:
+            return redirect(checkout_session.url)
+
+        return abort(500)
+
+    @bp.route("/disable-autorenew", methods=["POST"])
+    @authentication_required
+    def disable_autorenew() -> Response | str | Tuple[Response | str, int]:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                stripe.Subscription.modify(user.stripe_subscription_id, cancel_at_period_end=True)
+                user.stripe_subscription_cancel_at_period_end = True
+                db.session.add(user)
+                db.session.commit()
+
+                current_app.logger.info(
+                    f"Autorenew disabled for subscription {user.stripe_subscription_id} for user {user.id}"  # noqa: E501
+                )
+
+                flash("Autorenew has been disabled.")
+                return jsonify(success=True)
+            except stripe._error.StripeError as e:
+                current_app.logger.error(f"Stripe error: {e}")
+                return jsonify(success=False), 400
+
+        return jsonify(success=False), 400
+
+    @bp.route("/enable-autorenew", methods=["POST"])
+    @authentication_required
+    def enable_autorenew() -> Response | str | Tuple[Response | str, int]:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                stripe.Subscription.modify(user.stripe_subscription_id, cancel_at_period_end=False)
+                user.stripe_subscription_cancel_at_period_end = False
+                db.session.add(user)
+                db.session.commit()
+
+                current_app.logger.info(
+                    f"Autorenew enabled for subscription {user.stripe_subscription_id} for user {user.id}"  # noqa: E501
+                )
+
+                flash("Autorenew has been enabled.")
+                return jsonify(success=True)
+            except stripe._error.StripeError as e:
```

Comment:
Might want `exc_info=True` in the logger to get more details instead of `{e}`

Reply:
Fixed in 977935a15167450c085c93340cb31e0ac6efca01